### PR TITLE
Better channel write logic

### PR DESCRIPTION
**Pull request by ShanniLi**
r @anson627 

benchmark: 70k+ qps

### CODE REVIEW

File: tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java [hunk 1]
```diff
@@ -148,7 +128,15 @@ public void onSuccess(ResponseMessage response) {
 
                 // Since the request was handled, decrement the queued requests count
                 queuedRequests.decrementAndGet();
-                ctx.writeAndFlush(response);
+
+                // TODO: aggregate the flush
+                if (!ctx.channel().isWritable()) {
+                    synchronized (responseQueue) {
```
Comment (anson627) [inline tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java]: ConcurrentLinkedQueue is lock free, which might be faster?

File: tchannel-core/src/test/java/com/uber/tchannel/channels/ErrorResponseHandlingTest.java [hunk 2]
```diff
@@ -86,49 +86,6 @@ public void testBadRequestErrorOnInvalidArgScheme() throws Exception {
         client.shutdown();
     }
 
-    @Test
```
Comment (anson627) [inline tchannel-core/src/test/java/com/uber/tchannel/channels/ErrorResponseHandlingTest.java]: why delete a test?

Comment (ShanniLi) [inline tchannel-core/src/test/java/com/uber/tchannel/channels/ErrorResponseHandlingTest.java]: I removed the rate-limiting, so the test won't apply anymore.

**Comment by anson627** [comment]
:+1: any improvement to RPS after switching to a lock free data structure.

**Comment by ShanniLi** [comment]
Actually the new version is worse. After the change, the QPS can never reach 70k again. You can try it out using PingPongServerBenchmark.

**Pull request merged**